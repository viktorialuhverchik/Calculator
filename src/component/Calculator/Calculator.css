.calculator {
    max-width: 310px;
    display: flex;
    flex-direction: column;
    background-color: #ecf3f3;
    border: 1px solid #c2d3d3;
    border-radius: 30px;
    margin: 10vh auto;
    justify-content: center;
    align-items: center;
    box-shadow: 2px 2px 2px #c2d3d3;
    animation-name: calculator;
    animation-duration: 1s;
}

.buttons-wrapper {
    padding: 15px 0;
}

.row {
    display: flex;
    justify-content: center;
    margin: 2px auto;
}

.btn {
    height: 45px;
    width: 65px;
    margin-right: 2px;
    display: flex;
    align-items: center;
    justify-content: center;
    outline: none;
}

.btn:focus {
    outline: none;
    box-shadow: none;
}

.number {
    border: 1px solid #FF3D00;
    border-radius: 5px;
    font-size: 20px;
    font-weight: 500;
    color: #6e7575;
    transition: all 100ms;
}

.number:hover {
    color: #6e7575;
}

.number:active {
    color: #ecf3f3;
    font-size: 22px;
    background-color: #FF3D00;
    border: 1px solid #FF3D00;
}

.symbol {
    color: #FF3D00;
    font-weight: 400;
}

.symbol:hover {
    color: #FF3D00;
}

.symbol:active {
    font-size: 20px;
}

.large {
    font-size: 24px;
    font-weight: 300;
}

.large:active {
    font-size: 30px;
}

.equally-wrapper {
    height: 40px;
    width: 60px;
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 2px 3px;
}

.equally {
    height: 45px;
    width: 45px;
    background-color: #FF3D00;
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 100%;
    color: #ecf3f3;
    font-size: 20px;
    transition: all 90ms;
}

.equally:hover {
    color: #ecf3f3;
}

.equally:active {
    height: 50px;
    width: 50px;
    justify-content: center;
    color: #ecf3f3;
    font-size: 24px;
    background-color: #FF3D00;
    border: 1px solid #FF3D00;
}

.delete-icon {
    height: 13px;
    width: 16px;
}

.delete-icon:active {
    height: 15px;
    width: 19px;
}

@keyframes calculator {
    from {opacity: 0;}
    to {opacity: 1;}
}
