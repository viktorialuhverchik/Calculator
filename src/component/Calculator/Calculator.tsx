import React, { FC } from 'react';
import { useDispatch } from 'react-redux';
import { calculate, clearValue, deleteLast, handleClick, toggleShowHistory } from '../../redux/actions/actions';
import Display from '../Display/Display';
import Themes from '../Themes/Themes';
import { Numbers, Operations, PropsCalculator } from '../../types';

import './Calculator.css';

const Calculator: FC<PropsCalculator> = ({isShowHistory, history, theme}) => {

    const dispatch: any = useDispatch();

    return (
        <div className="calculator container">
            <Themes theme={theme} />
            <Display history={history}/>
            <div className="buttons-wrapper">
                <div className="row">
                    <button
                        className="btn symbol"
                        data-testid="clear"
                        onClick={() => dispatch(clearValue())}
                    >C
                    </button>
                    <button
                        className="btn delete-icon_button"
                        data-testid="delete-last"
                        onClick={() => dispatch(deleteLast())}
                    >
                        <img alt="delete" src="/icons/delete.png" className="delete-icon" />
                    </button>
                    <button
                        className="btn symbol"
                        data-testid="percent"
                        onClick={() => dispatch(handleClick(Operations.Percent))}
                    >%
                    </button>
                    <button
                        className="btn symbol large"
                        data-testid="division"
                        onClick={() => dispatch(handleClick(Operations.Division))}
                    >&#247;
                    </button>
                </div>

                <div className="row">
                    <button
                        className="btn number"
                        data-testid="seven"
                        onClick={() => dispatch(handleClick(Numbers.Seven))}
                    >7
                    </button>
                    <button
                        className="btn number"
                        data-testid="eight"
                        onClick={() => dispatch(handleClick(Numbers.Eight))}
                    >8
                    </button>
                    <button
                        className="btn number"
                        data-testid="nine"
                        onClick={() => dispatch(handleClick(Numbers.Nine))}
                    >9
                    </button>
                    <button
                        className="btn symbol large"
                        data-testid="multiplication"
                        onClick={() => dispatch(handleClick(Operations.Multiplication))}
                    >&#215;
                    </button>
                </div>

                <div className="row">
                    <button
                        className="btn number"
                        data-testid="four"
                        onClick={() => dispatch(handleClick(Numbers.Four))}
                    >4
                    </button>
                    <button
                        className="btn number"
                        data-testid="five"
                        onClick={() => dispatch(handleClick(Numbers.Five))}
                    >5
                    </button>
                    <button
                        className="btn number"
                        data-testid="six"
                        onClick={() => dispatch(handleClick(Numbers.Six))}
                    >6
                    </button>
                    <button
                        className="btn symbol large"
                        data-testid="minus"
                        onClick={() => dispatch(handleClick(Operations.Minus))}
                    >&#8722;
                    </button>
                </div>

                <div className="row">
                    <button
                        className="btn number"
                        data-testid="one"
                        onClick={() => dispatch(handleClick(Numbers.One))}
                    >1
                    </button>
                    <button
                        className="btn number"
                        data-testid="two"
                        onClick={() => dispatch(handleClick(Numbers.Two))}
                    >2
                    </button>
                    <button
                        className="btn number"
                        data-testid="three"
                        onClick={() => dispatch(handleClick(Numbers.Three))}
                    >3
                    </button>
                    <button
                        className="btn symbol large"
                        data-testid="plus"
                        onClick={() => dispatch(handleClick(Operations.Plus))}
                    >+
                    </button>
                </div>

                <div className="row">
                    <button
                        className="btn number"
                        data-testid="history"
                        onClick={() => dispatch(toggleShowHistory(isShowHistory))}>
                        <i className="fa fa-history" aria-hidden="true"></i>
                    </button>
                    <button
                        className="btn number"
                        data-testid="zero"
                        onClick={() => dispatch(handleClick(Numbers.Zero))}
                    >0
                    </button>
                    <button className="btn number"
                        data-testid="dot"
                        onClick={() => dispatch(handleClick(Operations.Dot))}
                    >.
                    </button>

                    <div className="equally-wrapper">
                        <div
                            className="btn equally"
                            data-testid="equally"
                            onClick={() => dispatch(calculate())}
                        >=
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Calculator;
