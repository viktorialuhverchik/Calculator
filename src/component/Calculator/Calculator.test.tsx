import React from 'react';
import { renderWithRedux } from '../../index.test';
import { fireEvent } from '@testing-library/react';
import Calculator from './Calculator';

let isShowHistory: boolean = false;
let history: string[] = ["1+2=3"];
let theme: string = "light";

describe('Calculator component',() => {

    it('renders Calculator component', () => {
        const wrapper = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('click clear value', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("clear"));
        expect(getByTestId("clear")).toBeTruthy();
    });

    it('click delete last', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("delete-last"));
        expect(getByTestId("delete-last")).toBeTruthy();
    });

    it('click handle click percent', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("percent"));
        expect(getByTestId("percent")).toBeTruthy();
    });

    it('click handle click division', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("division"));
        expect(getByTestId("division")).toBeTruthy();
    });

    it('click handle click seven', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("seven"));
        expect(getByTestId("seven")).toBeTruthy();
    });

    it('click handle click eight', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("eight"));
        expect(getByTestId("eight")).toBeTruthy();
    });

    it('click handle click nine', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("nine"));
        expect(getByTestId("nine")).toBeTruthy();
    });

    it('click handle click multiplication', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("multiplication"));
        expect(getByTestId("multiplication")).toBeTruthy();
    });

    it('click handle click four', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("four"));
        expect(getByTestId("four")).toBeTruthy();
    });

    it('click handle click five', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("five"));
        expect(getByTestId("five")).toBeTruthy();
    });

    it('click handle click six', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("six"));
        expect(getByTestId("six")).toBeTruthy();
    });

    it('click handle click minus', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("minus"));
        expect(getByTestId("minus")).toBeTruthy();
    });

    it('click handle click one', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("one"));
        expect(getByTestId("one")).toBeTruthy();
    });

    it('click handle click two', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("two"));
        expect(getByTestId("two")).toBeTruthy();
    });

    it('click handle click three', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("three"));
        expect(getByTestId("three")).toBeTruthy();
    });

    it('click handle click plus', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("plus"));
        expect(getByTestId("plus")).toBeTruthy();
    });

    it('click handle click history', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("history"));
        expect(getByTestId("history")).toBeTruthy();
    });

    it('click handle click zero', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("zero"));
        expect(getByTestId("zero")).toBeTruthy();
    });

    it('click handle click dot', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("dot"));
        expect(getByTestId("dot")).toBeTruthy();
    });

    it('click handle click equally', () => {
        const { getByTestId } = renderWithRedux(<Calculator isShowHistory={isShowHistory} history={history} theme={theme} />);
        fireEvent.click(getByTestId("equally"));
        expect(getByTestId("equally")).toBeTruthy();
    });
});
