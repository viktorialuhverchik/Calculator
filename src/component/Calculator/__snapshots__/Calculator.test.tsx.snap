// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders Calculator component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculator container"
      >
        <div
          class="themes"
        >
          <i
            aria-hidden="true"
            class="fa fa-moon-o"
            data-testid="moon"
          />
        </div>
        <div
          class="display-wrapper"
        >
          <div
            class="display-history"
          >
            <span
              class="display-last_value"
              data-testid="last-value"
            />
          </div>
          <div
            class="display-value"
          />
          <div
            class="display-equal"
          >
            = 0
          </div>
        </div>
        <div
          class="buttons-wrapper"
        >
          <div
            class="row"
          >
            <button
              class="btn symbol"
              data-testid="clear"
            >
              C
            </button>
            <button
              class="btn delete-icon_button"
              data-testid="delete-last"
            >
              <img
                alt="delete"
                class="delete-icon"
                src="/icons/delete.png"
              />
            </button>
            <button
              class="btn symbol"
              data-testid="percent"
            >
              %
            </button>
            <button
              class="btn symbol large"
              data-testid="division"
            >
              ÷
            </button>
          </div>
          <div
            class="row"
          >
            <button
              class="btn number"
              data-testid="seven"
            >
              7
            </button>
            <button
              class="btn number"
              data-testid="eight"
            >
              8
            </button>
            <button
              class="btn number"
              data-testid="nine"
            >
              9
            </button>
            <button
              class="btn symbol large"
              data-testid="multiplication"
            >
              ×
            </button>
          </div>
          <div
            class="row"
          >
            <button
              class="btn number"
              data-testid="four"
            >
              4
            </button>
            <button
              class="btn number"
              data-testid="five"
            >
              5
            </button>
            <button
              class="btn number"
              data-testid="six"
            >
              6
            </button>
            <button
              class="btn symbol large"
              data-testid="minus"
            >
              −
            </button>
          </div>
          <div
            class="row"
          >
            <button
              class="btn number"
              data-testid="one"
            >
              1
            </button>
            <button
              class="btn number"
              data-testid="two"
            >
              2
            </button>
            <button
              class="btn number"
              data-testid="three"
            >
              3
            </button>
            <button
              class="btn symbol large"
              data-testid="plus"
            >
              +
            </button>
          </div>
          <div
            class="row"
          >
            <button
              class="btn number"
              data-testid="history"
            >
              <i
                aria-hidden="true"
                class="fa fa-history"
              />
            </button>
            <button
              class="btn number"
              data-testid="zero"
            >
              0
            </button>
            <button
              class="btn number"
              data-testid="dot"
            >
              .
            </button>
            <div
              class="equally-wrapper"
            >
              <div
                class="btn equally"
                data-testid="equally"
              >
                =
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator container"
    >
      <div
        class="themes"
      >
        <i
          aria-hidden="true"
          class="fa fa-moon-o"
          data-testid="moon"
        />
      </div>
      <div
        class="display-wrapper"
      >
        <div
          class="display-history"
        >
          <span
            class="display-last_value"
            data-testid="last-value"
          />
        </div>
        <div
          class="display-value"
        />
        <div
          class="display-equal"
        >
          = 0
        </div>
      </div>
      <div
        class="buttons-wrapper"
      >
        <div
          class="row"
        >
          <button
            class="btn symbol"
            data-testid="clear"
          >
            C
          </button>
          <button
            class="btn delete-icon_button"
            data-testid="delete-last"
          >
            <img
              alt="delete"
              class="delete-icon"
              src="/icons/delete.png"
            />
          </button>
          <button
            class="btn symbol"
            data-testid="percent"
          >
            %
          </button>
          <button
            class="btn symbol large"
            data-testid="division"
          >
            ÷
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="btn number"
            data-testid="seven"
          >
            7
          </button>
          <button
            class="btn number"
            data-testid="eight"
          >
            8
          </button>
          <button
            class="btn number"
            data-testid="nine"
          >
            9
          </button>
          <button
            class="btn symbol large"
            data-testid="multiplication"
          >
            ×
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="btn number"
            data-testid="four"
          >
            4
          </button>
          <button
            class="btn number"
            data-testid="five"
          >
            5
          </button>
          <button
            class="btn number"
            data-testid="six"
          >
            6
          </button>
          <button
            class="btn symbol large"
            data-testid="minus"
          >
            −
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="btn number"
            data-testid="one"
          >
            1
          </button>
          <button
            class="btn number"
            data-testid="two"
          >
            2
          </button>
          <button
            class="btn number"
            data-testid="three"
          >
            3
          </button>
          <button
            class="btn symbol large"
            data-testid="plus"
          >
            +
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="btn number"
            data-testid="history"
          >
            <i
              aria-hidden="true"
              class="fa fa-history"
            />
          </button>
          <button
            class="btn number"
            data-testid="zero"
          >
            0
          </button>
          <button
            class="btn number"
            data-testid="dot"
          >
            .
          </button>
          <div
            class="equally-wrapper"
          >
            <div
              class="btn equally"
              data-testid="equally"
            >
              =
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
